rest_framework-stubs/mixins.pyi

```diff
@@ -1,13 +1,17 @@
-from typing import Any, Dict
+from typing import Any, Dict, TypeVar
 
 from django.db.models import Model
+
+from rest_framework.generics import UsesQuerySet
 from rest_framework.request import Request
 from rest_framework.response import Response
 from rest_framework.serializers import BaseSerializer
 
+_MT = TypeVar("_MT", bound=Model)
+
 class CreateModelMixin:
     def create(self, request: Request, *args: Any, **kwargs: Any) -> Response: ...
-    def perform_create(self, serializer: BaseSerializer) -> None: ...
+    def perform_create(self: UsesQuerySet[_MT], serializer: BaseSerializer[_MT]) -> None: ...
```

Comment:
Wow, I didn't know you could do that 😮 nice.

Reply:
@intgr mypy has a section on ["Advanced self-types"](https://mypy.readthedocs.io/en/stable/more_types.html#advanced-uses-of-self-types). Pretty cool feature! 👍